dropshot_endpoint/src/lib.rs

```diff
@@ -286,6 +290,32 @@ fn do_endpoint(
         })
         .collect::<Vec<_>>();
 
+    // We want to construct a function that will call the user's endpoint, so
+    // we can check the future it returns for bounds that otherwise produce
+    // inscrutable error messages (like returning a non-`Send` future). We
+    // produce a wrapper function that takes all the same argument types,
+    // which requires building up a list of argument names.
```

Comment:
why can't we use the names from the original function?

Reply:
It will break if they used `_` as one or more argument names. Not likely, but certainly legal.